### PR TITLE
Update Kuttl tests to better handle variable error output

### PR DESCRIPTION
**Pull request by tjmoore4**
Update Kuttl tests to better handle variable error output.

### CODE REVIEW

**Review by benjaminjb — APPROVED** [review]
GCP auth sometimes adds its own statements to returns/print statements, so this looks like a fair way to check returns with some slack for that

File: testing/kuttl/e2e/backup/11--backup-with-new-flag.yaml [hunk 1]
```diff
@@ -12,10 +12,7 @@ commands:
       exit 1
     fi
 
-    # FIXME(cbandy): why does the output start with a blank line?
-    ERROR=$(echo "${RESULT}" | head -n2)
```
Comment (cbandy) [inline testing/kuttl/e2e/backup/11--backup-with-new-flag.yaml:16]: 🤔 It seems like I was trying to get just the first line, but I don't recall why. Is there something else that normally prints on later lines?

**Review by cbandy — APPROVED** [review]
I would like it if our tests could be more focused/specific about the errors _we_ return, but I suppose with client-go doing whatever it wants to stderr we're kinda stuck.

File: testing/kuttl/e2e/backup/11--backup-with-new-flag.yaml (hunk 1 above)
Comment (tjmoore4) [reply testing/kuttl/e2e/backup/11--backup-with-new-flag.yaml:16]: From my tests, it looks like it normally prints the error this way:
```
Error requesting update: Apply failed with 1 conflict: conflict with "kubectl-kuttl" using postgres-operator.crunchydata.com/v1beta1: .spec.backups.pgbackrest.manual.repoName
Error: Apply failed with 1 conflict: conflict with "kubectl-kuttl" using postgres-operator.crunchydata.com/v1beta1: .spec.backups.pgbackrest.manual.repoName
```
The underlying error is the same, but the first time starts with `Error requesting update` and the second line starts with `Error`.

**Pull request merged**